Classes/Dataset.cs

```diff
@@ -18,11 +19,17 @@ public class Dataset: IEquatable<Dataset>
         public string Key { get; set; } //Schlüssel, der in Datenbank hinterlegt ist
         public ConcurrentQueue<string> ids; //Asynchrone Queue
 
-        async Task DownloadFilesAsync(string path)
+        public class counterHolder
```

Comment:
Wird die Klasse später noch erweitert/in einem anderen Kontext weitergenutzt? SOnmst müssten wir sehen, dass wir ihren Member irgendwo anders implementieren

Reply:
Ich musste den COunter so einsetzen, da ich in async-Funktionen keine ref int übergeben kann, deswegen muss ich da die Klasse übergeben oder ncoh eine übergeordnetete Variable definieren, deswegen die nested-class

Reply:
Wird dementsprechend nicht weiter genutzt